src/Courses/TopicSettings/ProblemSettings.tsx

```diff
@@ -143,13 +155,117 @@ export const ProblemSettings: React.FC<ProblemSettingsProps> = ({selected, setSe
+                                InputProps={{
+                                    startAdornment: <InputAdornment position='start'>
+                                        <Tooltip title='Randomize'>
+                                            <IconButton
+                                                aria-label='reload problem with a random seed'
+                                                onClick={(event: any)=>{
+                                                    event.stopPropagation();
+                                                    setPreviewSettings(settings => ({...settings, seed: _.random(0, 999999, false)}));
+                                                    forceUpdate();
+                                                    cycleFlipped();
+                                                }}
+                                            >
+                                                <motion.div animate={{scaleX: flipped}} ><FaDice/></motion.div>
+                                            </IconButton>
+                                        </Tooltip>
+                                    </InputAdornment>,
+                                    endAdornment: <InputAdornment position='end'>
+                                        <Tooltip title='Reload'>
+                                            <IconButton
+                                                aria-label='reload problem with current seed'
+                                                onClick={(event: any)=>{
+                                                    event.stopPropagation();
+                                                    controls.start({rotate: 360, transition: {duration: 0.5}});
+                                                    forceUpdate();
+                                                }}
+                                            >
+                                                <motion.div animate={controls}><Refresh /></motion.div>
+                                            </IconButton>
+                                        </Tooltip>
+                                    </InputAdornment>
+                                }}
+                            />
+                        </Grid>
+                        <Grid xs={12} item>
+                            {pgRegex.test(previewSettings.path) ?
+                                <ProblemIframe 
+                                    problem={forcedUpdate} 
+                                    previewPath={previewSettings.path}
+                                    previewSeed={previewSettings.seed}
+                                    setProblemStudentGrade={() => {}}
+                                    readonly={false} /> :
+                                <>
+                                    {previewSettings.path !== '' && (
+                                        <Alert variant='warning'>The path you&apos;ve entered is invalid.</Alert>
```

Comment:
I personally prefer avoiding apostrophes because it makes it harder to search

If I see `you're` that's what I might do a global search for and also I feel like it's more formal to do `you have`